cmd/thanos/store.go

```diff
@@ -83,7 +83,7 @@ func (sc *storeConfig) registerFlag(cmd extkingpin.FlagClause) {
 	sc.grpcConfig = *sc.grpcConfig.registerFlag(cmd)
 
 	cmd.Flag("data-dir", "Local data directory used for caching purposes (index-header, in-mem cache items and meta.jsons). If removed, no data will be lost, just store will have to rebuild the cache. NOTE: Putting raw blocks here will not cause the store to read them. For such use cases use Prometheus + sidecar.").
-		Default("./data").StringVar(&sc.dataDir)
+		Default("").StringVar(&sc.dataDir)
```

Comment:
We shouldn't change the default.

Reply:
> We shouldn't change the default.

If the default value is not "", users can not set the `data-dir` to "".

Reply:
I don't think it is reasonable since it changes the default behavior. In this case please add a separate flag.

Reply:
A flag `--disable-caching-index-header-file` was added to disable caching index header file.